samples/mac/weather/MyManagedStuff/MyManagedStuff/WeatherFetcher.cs

```diff
@@ -0,0 +1,50 @@
+﻿﻿using System;
+using System.Json;
+using System.Net;
+
+namespace MyManagedStuff {
+	public class WeatherFetcher {
+
+		static string urlTemplate = @"https://query.yahooapis.com/v1/public/yql?q=select%20item.condition%20from%20weather.forecast%20where%20woeid%20in%20(select%20woeid%20from%20geo.places(1)%20where%20text%3D%22{0}%2C%20{1}%22)&format=json&env=store%3A%2F%2Fdatatables.org%2Falltableswithkeys";
+		public string City { get; private set; }
+		public string State { get; private set; }
+
+
+		public WeatherFetcher (string city, string state)
+		{
+			City = city;
+			State = state;
+		}
+
+		public WeatherResult GetWeather ()
+		{
+			try {
+				using (var wc = new WebClient ()) {
+					var url = string.Format (urlTemplate, City, State);
+					var str = wc.DownloadString (url);
+					var json = JsonValue.Parse (str)["query"]["results"]["channel"]["item"]["condition"];
+					var result = new WeatherResult (json["temp"], json["text"]);
+					return result;
+				}
+			}
+			catch (Exception ex) {
+				Console.WriteLine (ex.Message);
+				Console.WriteLine (ex?.InnerException?.Message);
+				Console.WriteLine (ex?.InnerException?.InnerException?.Message);
+				throw ex;
```

Comment:
We do not support exceptions, because it's quite lacking in ObjC, so we should not promote it's use in the samples. Even more since a network error is not exceptional ;-)

Maybe replace the `throw ex;` by `return null;` and put a comment on the ObjC code ?